packages/api/src/router/widgets/calendar.ts

```diff
@@ -0,0 +1,19 @@
+import type { CalendarEvent } from "@homarr/integrations/types";
+import { createCacheChannel } from "@homarr/redis";
+
+import { createManyIntegrationMiddleware } from "../../middlewares/integration";
+import { createTRPCRouter, publicProcedure } from "../../trpc";
+
+export const calendarRouter = createTRPCRouter({
+  findAllEvents: publicProcedure
+    .unstable_concat(createManyIntegrationMiddleware("sonarr", "radarr", "readarr", "lidarr"))
+    .query(async ({ ctx }) => {
+      const data = ctx.integrations.map(async (integration) => {
+        const cache = createCacheChannel<CalendarEvent[]>(`calendar:${integration.id}`);
```

Comment:
I would suggest to move the creation of this cache channel to the redis package

Reply:
What exactly should I move? Can you elaborate

Reply:
Move 
```ts
export const createCalendarIntegrationChannel = (integrationId: string) => createCacheChannel<CalendarEvent[]>(`calendar:${integrationId}`);
```
to the redis package like the other channels we export there. Otherwise once the structure of the path changes it we could forget about one of them